code/__DEFINES/logging.dm

```diff
@@ -16,12 +16,29 @@
 #define INVESTIGATE_EXONET			"exonet"
 #define INVESTIGATE_CIRCUIT			"circuit"
 
-//Individual logging defines
-#define INDIVIDUAL_ATTACK_LOG		"Attack log"
-#define INDIVIDUAL_SAY_LOG			"Say log"
-#define INDIVIDUAL_EMOTE_LOG		"Emote log"
-#define INDIVIDUAL_OOC_LOG			"OOC log"
-#define INDIVIDUAL_OWNERSHIP_LOG	"Ownership log"
-#define INDIVIDUAL_SHOW_ALL_LOG		"All logs"
+// Logging types for log_message()
+#define LOG_ATTACK		(1 << 0)
+#define LOG_SAY			(1 << 1)
+#define LOG_WHISPER		(1 << 2)
+#define LOG_EMOTE		(1 << 3)
+#define LOG_DSAY		(1 << 4)
+#define LOG_PDA			(1 << 5)
+#define LOG_CHAT		(1 << 6)
+#define LOG_COMMENT		(1 << 7)
+#define LOG_TELECOMMS	(1 << 8)
+#define LOG_OOC			(1 << 9)
+#define LOG_ADMIN		(1 << 10)
+#define LOG_OWNERSHIP	(1 << 11)
+#define LOG_GAME		(1 << 12)
```

Comment:
my concern with bitflags is if we have too many types in the future to use bitflags but eh I guess investigate exists.

Reply:
We're only using 13 of the 32 bits at the moment, so I think it should be fine for the time being. It should be easy enough to switch to string constants and lists if we run out.

Reply:
Yeah this will be fine

Reply:
FYI we can only use 24 bits

Reply:
Huh, the more you know. Thanks BYOND.